src/components/atoms/formInput/TextInput.tsx

```diff
@@ -0,0 +1,22 @@
+import { TextField, TextFieldProps } from "@mui/material";
+import { Input } from "../../../common/constants/enums";
+import { kurtTheme } from "../../../theme";
+import { inputToFormProps } from "../../../common/constants/constants";
+
+export type TextInputProps = {
+  fieldName: Input
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
```

Comment:
❓ Is this just to make the `onChange` prop of `TextFieldProps` required?

Reply:
Good point, kind of was to start. But I feel like we probably don't need it to be required.

Reply:
https://github.com/mwkwsd/sensensomething/pull/56/files

Reply:
Cool. Definitely will want to walk through that at some point. `[P in K]-?: T[P]` looks like literally nothing to me.

Reply:
That's the idea.
Or... wait
do you mean it looks like it does nothing
Or do you mean you don't understand it?

Reply:
Agreed with Ken. 
I'd love an explanation of that code. I've seen generics before but this is totally new code to me as well.

Reply:
I mean I don't understand it.

Reply:
I started to write something up about this. Then it got too much. Happy to go over it in detail if you want. Basically, this util type turns the keys in T provided in K required on T